### ISSUE TITLE
Unable to change timeout value in job_template endpoint as a template admin

### ISSUE THREAD

**Issue by msilmser** [ansible/awx#2798]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - API

##### SUMMARY
Users with the job template admin role are unable to change the "timeout" value at the job template endpoint. 

##### ENVIRONMENT
* Tower version: 3.3

##### STEPS TO REPRODUCE
1. Create a user with job template admin permission
2. Navigate to /api/v2/job_templates/N
3. Attempt to patch the timeout value

##### EXPECTED RESULTS
Able to change the timeout value

##### ACTUAL RESULTS
Permissions error:
```{
    "detail": "You do not have permission to perform this action."
}```

**Comment by AlanCoding** [comment]
To change sensitive fields in a job template, you need permission to the related project and inventory.

If you would like a better error message, see the proposal at https://github.com/ansible/awx/pull/2415, which still doesn't have any traction.

**Comment by wenottingham** [comment]
Why is timeout a sensitive field?

**Comment by AlanCoding** [comment]
I cannot say. I believe it is not allowed because it is not in this list:

https://github.com/ansible/awx/blob/50b0a5a54d443c4b9a6378ade8371cfd2a6187fe/awx/main/access.py#L1387-L1392

**Comment by wenottingham** [comment]
Then I think we should just add it to that list, if that's all that it takes. It's certainly less sensitive than `diff_mode` is.

**Comment by aw-huit** [comment]
I can see why you need permission for the project, but I can't figure out the reasoning behind needing permission for inventories as well.  I always thought templates don't have to be tied to any specific inventory.

**Comment by wenottingham** [comment]
Fixed in #2848.
